core/trino-main/src/main/java/io/trino/execution/QueryTracker.java

```diff
@@ -139,7 +139,7 @@ public void stop()
 
     public Collection<T> getAllQueries()
     {
-        return ImmutableList.copyOf(queries.values());
+        return unmodifiableCollection(queries.values());
```

Comment:
why does it matter? Is that used in any performance critical context? 
If not it would be nicer to have returned collection "frozen"

Reply:
I saw that on the head dump. This copy is unnecessary.

Reply:
And this method is called over and over again, for example when enforcing resource limits